anasymod/templates/zynq_gpio.py

```diff
@@ -15,36 +23,41 @@ def __init__(self, ps_vlnv='xilinx.com:ip:processing_system7:5.5',
         self.i_data = DigitalSignal(name='i_data', width=32, abspath=None)
 
         # generate text
-        self.text = f'''
+
+    TEMPLATE_TEXT = '''\
 ############################
 # Create the block diagram #
 ############################
 
 # Initialize
 
-create_bd_design "{self.design_name}"
+create_bd_design "{{subst.design_name}}"

```

Comment:
I switched to using the Jinja template style, so single curly braces are replaced with double, ``self`` is replaced with ``subst``, etc.